cluster/raft_cluster_endpoints.go

```diff
@@ -25,6 +26,43 @@ func (s *Raft) LeaderWithID() (string, string) {
 	return string(addr), string(id)
 }
 
+// StorageCandidates return the nodes in the raft configuration or memberlist storage nodes
+// based on the current configuration of the cluster if it does have  MetadataVoterOnly nodes.
+func (s *Raft) StorageCandidates() []string {
+	if s.store.raft == nil {
+		// get candidates from memberlist
+		return s.nodeSelector.StorageCandidates()
```

Comment:
In which situation will this ever be valid ? Wondering if we should just error instead.

Reply:
good point, i made it as percussion, in case of long restore going on and a request came, however thinking again about i think by that time nodes shouldn't report ready anyway

but that will require the interface change

Reply:
What i'm afraid is that our behaviour to select storage node will be different depending on whether or not raft is online. Maybe instead we should find a way to wait and/or enable to check for storage node until raft comes up ? Not sure of the scale of the change this is fine and can stay for now if we prefer.

Reply:
Hmm, i think the behavior would be the same because memebrlist.StorageCandidates() are the source of truth if raft configuration servers are not in memberlist storage candidates we will fall back to memberlist 

So the behavior always fallback to memberlist candidates if there is something wrong which is the original implementation. The only case we need to catch if one of the nodes exist in RAFT but down in memberlist

Reply:
i will try to get an e2e test in place to cover that case though